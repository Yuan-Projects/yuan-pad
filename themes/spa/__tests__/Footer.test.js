import renderer from 'react-test-renderer';
import Footer from '../src/footer';
import NormalUser from '../__mocks__/user-regular';
import GuestUser from '../__mocks__/user-guest';
import AdminUser from '../__mocks__/user-admin';
import SiteConfig from '../__mocks__/app-config1';
import LangEn from '../__mocks__/site-translate-en';

it('Footer renders correctly', () => {
  const propsNormal = {
    user: NormalUser,
    appConfig: SiteConfig,
    lang: LangEn
  };
  const treeNormal = renderer.create(<Footer {...propsNormal} />).toJSON();
  expect(treeNormal).toMatchSnapshot();

  const propsGuest = {
    user: GuestUser,
    appConfig: SiteConfig,
    lang: LangEn
  };
  const treeGuest = renderer.create(<Footer {...propsGuest} />).toJSON();
  expect(treeGuest).toMatchSnapshot();

  const propsAdmin = {
    user: AdminUser,
    appConfig: SiteConfig,
    lang: LangEn
  };
  const treeAdmin = renderer.create(<Footer {...propsAdmin} />).toJSON();
  expect(treeAdmin).toMatchSnapshot();
})