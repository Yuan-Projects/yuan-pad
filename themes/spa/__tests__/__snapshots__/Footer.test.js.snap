// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <p>
    Copyright © MySite
     
    <a
      href="mailto:rainyjune@live.cn"
    >
      Email
    </a>
     
    <span
      dangerouslySetInnerHTML={
        {
          "__html": "",
        }
      }
    />
  </p>
  <p>
    Powered by 
    <a
      href="https://github.com/rainyjune/yuan-pad"
      target="_blank"
      title="Find More"
    >
      YuanPad
    </a>
     
    <a
      href="index.php?action=rss"
    >
      <img
        alt="rss"
        src="misc/images/rss-icon.png"
      />
    </a>
  </p>
</footer>
`;

exports[`Footer renders correctly 2`] = `
<footer>
  <p>
    Copyright © MySite
     
    <a
      href="mailto:rainyjune@live.cn"
    >
      Email
    </a>
     
    <span
      dangerouslySetInnerHTML={
        {
          "__html": "",
        }
      }
    />
  </p>
  <p>
    Powered by 
    <a
      href="https://github.com/rainyjune/yuan-pad"
      target="_blank"
      title="Find More"
    >
      YuanPad
    </a>
     
    <a
      href="index.php?action=rss"
    >
      <img
        alt="rss"
        src="misc/images/rss-icon.png"
      />
    </a>
  </p>
</footer>
`;

exports[`Footer renders correctly 3`] = `
<footer>
  <p>
    Copyright © MySite
     
    <a
      href="mailto:rainyjune@live.cn"
    >
      Email
    </a>
     
    <span
      dangerouslySetInnerHTML={
        {
          "__html": "<a href='index.php?action=control_panel'>Admin</a>",
        }
      }
    />
  </p>
  <p>
    Powered by 
    <a
      href="https://github.com/rainyjune/yuan-pad"
      target="_blank"
      title="Find More"
    >
      YuanPad
    </a>
     
    <a
      href="index.php?action=rss"
    >
      <img
        alt="rss"
        src="misc/images/rss-icon.png"
      />
    </a>
  </p>
</footer>
`;
